### DISCUSSION TITLE
Vibration Interval

### DISCUSSION THREAD

**Discussion by Cotrox** [zepp-health/zeppos-forum#69] [Q&A]
Hi guys, I want to realize a sort of metronome app with the ability to set a time in ms and use it when you are playing instruents. Your watch have to be setted up with an interval in MS and has to vibrate every XXXXms .

I tried with:

setInterval(() => {
      const vibrate = hmSensor.createSensor(hmSensor.id.VIBRATE);
      vibrate.start();
      vibrate.setMode(VIBRATOR_SCENE_DURATION_LONG);
      vibrate.start();
    }, 1000)

But it only vibrate once!

How can i do it?

**Comment by juan518munoz** [comment]
Use the `createTimer` function for that

```
//Create timer, delay 500ms to trigger, then execute every 1000ms.
const timer1 = timer.createTimer(
  500,
  1000,
  function (option) {
    //callback
    console.log('timer callback')
    console.log(option.hour)
  },
  { hour: 0, minute: 15, second: 30 }
)

//Stop timer1
timer.stopTimer(timer1)
```

https://docs.zepp.com/docs/1.0/watchface/api/timer/createTimer/

**Reply by Cotrox** [reply]
mmmm i tried before but probably i make something wrong. Is there an easy wat to teast app on my pc to use console?

Anyway i tried this: but still not working...

`export const BUTTON_PARAMS = {
  x: (480 - 400) / 2,
  y: 240,
  w: DEVICE_WIDTH - 42 * 2,
  h: 100,
  radius: 30,
  normal_color: 0x71E0A9,
  press_color: 0x004d40,
  text: 'Start',
  text_size: 36,
  color: 0x000000,
  click_func: (button_widget) => {
    
    const interval = timer.createTimer(
      10,
      1000,
      function (option) {
        const vibrate = hmSensor.createSensor(hmSensor.id.VIBRATE);
        vibrate.start();
        vibrate.setMode(VIBRATOR_SCENE_DURATION_LONG);
        vibrate.start();
        //console.log(option.hour)
      },
      { hour: 0, minute: 15, second: 30 }
    )

    //timer.stopTimer(interval)
  }
}
`

**Comment by MyoungXUE** [comment]
Hi, please refer to the sample code of the vibration sensor, it is recommended to call the `stop` API first and then `start`.

https://docs.zepp.com/docs/1.0/reference/device-app-api/hmSensor/sensorId/VIBRATE/

```js
const vibrate = hmSensor.createSensor(hmSensor.id.VIBRATE)

function click() {
  vibrate.stop()
  vibrate.scene = 25
  vibrate.start()
}

click()

Page({
  onDestroy() {
    vibrate && vibrate.stop()
  }
})
```

**Reply by Cotrox** [reply]
Nothing, i'm also trying this solution as you can see there:

`const vibrate = hmSensor.createSensor(hmSensor.id.VIBRATE);

    const interval = timer.createTimer(
      10,
      1000,
      (option) => {
        vibrate.stop();
        vibrate.scene = 25;
        vibrate.start();

        /*Page({
          onDestroy() {
            vibrate && vibrate.stop();
          }
        })*/

        //console.log(option.hour)
      },
      {
        hour: 0,
        minute: 0,
        second: 0
      }
    )

    //timer.stopTimer(interval)`

**Reply by Cotrox** [reply]
**EDIT:** Now it seams to work. But not always... the interval is not continuous. It works but not with a perfect interval. IDKW

**Reply by Cotrox** [reply]
EDIT: It seams that interval stop when i move my hand and restart when i'm not in movements. It also stops when screen go in stand by. I think we need a way to let screen stay woke up :3

**Reply by juan518munoz** [reply]
Search for
```  hmApp.setScreenKeep(true);```
On documentation

**Reply by Cotrox** [reply]
Thanķ you! I was searching for it!

Anyway now i'm having these two problems... you can see the Repo here: [Click](https://github.com/[Cotrox/Repository-GitHub-ZeppOS-Apps](https://github.com/Cotrox/Repository-GitHub-ZeppOS-Apps))

- Text Content not showing appName value but shows the text 'appName'... it was right before...
- Interval stop works sometimes. For exemple if you let your hand now in movements.

**Reply by juan518munoz** [reply]
Be sure you are importing the i18n file correctly, it seems wrongly done.
https://github.com/Cotrox/Repository-GitHub-ZeppOS-Apps/blob/61ac78853ac31dd91a631ec95f744108b962ada6/Metronome/pages/gts4-mini/home/index.params.js#L1

The vibration object needs to be destroyed uppon exiting the app
https://github.com/juan518munoz/ZeppOS-Spotify/blob/d89a9f79bb6d49428ac07b4dda0bd7c986cce219/pages/home/index.page.js#L212

**Reply by Cotrox** [reply]
I just tried to change them but it doesn't works >_<
